### ISSUE TITLE
Training dynamics of the prior transformer

### ISSUE THREAD

**Issue by cdjhz** [openai/DALL-E#16]
I noticed that a footnote of the paper says that the dVAE model is underfitted and thus the codes are sampled with argmax. So does that mean the distribution over the code vocabulary is flatter rather than sharper, i.e., less confident? Also, I am wondering what is the final nll of the prior transformer on the valid set?

**Comment by adityaramesh** [comment]
The footnote was about the transformer prior model underfitting the dataset, not the dVAE. We have two choices when training the prior model: (1) sample from the categorical distribution given by the logits of the dVAE encoder, and (2) take the argmax. Option (1) would correctly optimize the KL term of the ELB given in Equation 1, but basically adds noise to the targets of the transformer and is kind of like label smoothing. So I would expect it to require make its task harder and require more capacity to get to the same level of sample quality in the underfitting regime. That's why we went with option (2).

On a smaller dataset for which the prior model is liable to overfit, option (1) might be preferable. I forgot the exact number, but I think the final validation loss for the transformer was something like ~3.6--3.7 nats/dim.

**Comment by cdjhz** [comment]
Thanks for your reply! I am training a similar vq-vae with a prior transformer modeling the latent codes on texts. However with K=512 and length=64, the transformer model can only fit the latent codes to around nll=6.0 on the validation set (also on the train set). Is it more likely due to the quality of the sampled codes (I use argmax. The vae encoder predicts codes with fairly high confidence and the average nll is below 1.0) or to the training setting of the transformer (I tuned a little bit but did not find a significant difference)? The number of examples is around 100k. It will be really helpful if you can give me some hints. Looking forward to your reply!
